ui/src/lib.rs

```diff
@@ -640,6 +640,37 @@ impl Tui {
         let max_i = memory.len() / 32;
         let min_len = format!("{:x}", max_i * 32).len();
 
+        // color memory words based on write/read
+        let (mut word, mut color) =
+            if let Instruction::OpCode(op) = debug_steps[current_step].instruction {
+                let stack_len = debug_steps[current_step].stack.len();
+                if stack_len > 0 {
+                    let w = debug_steps[current_step].stack[stack_len - 1];
+                    match op {
+                        0x51 => (Some(w.as_usize() / 32), Some(Color::Cyan)), // read mem word
+                        0x52 => (Some(w.as_usize() / 32), Some(Color::Red)),  // write mem word
```

Comment:
optionally use the revm::opcode constants here instead

Reply:
Didn't want to import it in Cargo.toml just for this but probably will want it for more things in the future so will do